### PR TITLE
Use ACLs for system functions

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/108/

File: internal/auth/auth.go [hunk 1]
```diff
@@ -130,11 +156,11 @@ func (a *Authorizer) aclForOp(ctx context.Context, op bakery.Op) (acl []string,
 		}
 		switch op.Action {
 		case ActionRead:
-			// Only admins are allowed to read global information.
-			return AdminACL, true, nil
+			acl, err := a.aclManager.ACL(ctx, readUserACL)
```
Comment (rogpeppe) [inline internal/auth/auth.go]: This seems a bit repetitive. How about adding a helper function at the start?

    managerACL := func(aclName string) ([]string, bool, err) {
        acl, err := a.aclManager.ACL(ctx, aclName)
        return acl, false, errgo.Mask(err)
    }
    switch kind {
    case ActionDischargeFor:
      return managerACL(dischargeForUserACL)
    etc

Alternatively, set an ACL name variable and have the switch fall through to the end.

File: internal/identity/server.go [hunk 2]
```diff
@@ -66,13 +70,41 @@ func New(sp ServerParams, versions map[string]NewAPIHandlerFunc) (*Server, error
...
+	if err != nil {
+		return nil, errgo.Mask(err)
+	}
+
+	aclAuthenticator := httpauth.New(oven, auth)
+	aclHandler := aclManager.NewHandler(aclstore.HandlerParams{
+		RootPath: "/acls",
+		Authenticate: func(ctx context.Context, w http.ResponseWriter, req *http.Request) (aclstore.Identity, error) {
+			ai, err := aclAuthenticator.Auth(ctx, req, identchecker.LoginOp)
+			if herr, ok := errgo.Cause(err).(*httpbakery.Error); ok {
```
Comment (rogpeppe) [inline internal/identity/server.go]: I think you need to write the response always. How about this?

      ai, err := aclAuthenticator.Auth(ctx, req, identchecker.LoginOp)
      if err != nil {
           WriteError(w, err)
           return errgo.Mask(err)
      }
      return ai.Identity.(aclstore.Identity), nil

File: internal/auth/auth.go (hunk 1 above)
Comment (mhilton) [reply internal/auth/auth.go]: There are a number of places where we add the username to the returned ACL. I'm inclined to leave this.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/110/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/111/

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**